db/migrate/20170529142557_unify_container_definition_and_container.rb

```diff
@@ -0,0 +1,169 @@
+    add_column :containers, :cpu_cores,          :float
+    add_column :containers, :container_group_id, :bigint
+    add_column :containers, :privileged,         :boolean
+    add_column :containers, :run_as_user,        :bigint
+    add_column :containers, :run_as_non_root,    :boolean
+    add_column :containers, :capabilities_add,   :string
+    add_column :containers, :capabilities_drop,  :string
+    add_column :containers, :command,            :text
+
+    containers = Arel::Table.new(:containers)
+    definitions = Arel::Table.new(:container_definitions)
+    say_with_time("Copying over columns from container_definition to container") do
+      %w(image image_pull_policy memory cpu_cores container_group_id privileged
+         run_as_user run_as_non_root capabilities_add capabilities_drop command).each do |column|
+        join_sql = definitions.project(definitions[column.to_sym])
+                              .where(definitions[:id].eq(containers[:container_definition_id])).to_sql
+        Container.update_all("#{column} = (#{join_sql})")
+      end
+    end
+
+    say_with_time("switch container_definition_id with container_id for container_port_configs") do
+      port_configs = Arel::Table.new(:container_port_configs)
+      join_sql = containers.project(containers[:id])
+                           .where(containers[:container_definition_id].eq(port_configs[:container_definition_id])).to_sql
+      ContainerPortConfig.update_all("container_definition_id = (#{join_sql})")
+    end
+
+    say_with_time("switch container_definition_id with container_id for container_port_configs") do
+      env_vars = Arel::Table.new(:container_env_vars)
+      join_sql = containers.project(containers[:id])
+                           .where(containers[:container_definition_id].eq(env_vars[:container_definition_id])).to_sql
+      ContainerEnvVar.update_all("container_definition_id = (#{join_sql})")
+    end
+
+    say_with_time("switch container_definition_id with container_id for security_contexts") do
+      security_contexts = Arel::Table.new(:security_contexts)
+      join_sql = containers.project(containers[:id])
+                           .where(containers[:container_definition_id].eq(security_contexts[:resource_id])
+                           .and(security_contexts[:resource_type].eq(Arel::Nodes::Quoted.new('ContainerDefinition')))).to_sql
+      SecurityContext.where(:resource_type => 'ContainerDefinition').update_all("resource_type = 'Container', resource_id = (#{join_sql})")
+    end
+
+    # relationships
+    rename_column :container_port_configs, :container_definition_id, :container_id
+    rename_column :container_env_vars, :container_definition_id, :container_id
+
+    remove_column :containers, :container_definition_id
+    drop_table :container_definitions
+  end
+
+  def down
+    create_table :container_definitions do |t|
+      t.belongs_to :ems, :type => :bigint
+      t.string     :ems_ref
```

Comment:
definition ems_ref is lost in `up`.  Would copying it back FROM containers restore old situation?
parser synthesizes `"#{pod_id}_#{container_def.name}_#{container_def.image}"`,
for containers it synthesizes `"#{pod_id}_#{container.name}_#{container.image}"`.
- [x] pod_id is same `pod.metadata.uid` for both.
- [x] name should be same, it's what parser cross-ref'd containers to container_definitions.
- [ ] is image necessarily same?  cc @enoodle 

(I don't know if 100% reversible migration is a blocker here, but want to understands things first.)